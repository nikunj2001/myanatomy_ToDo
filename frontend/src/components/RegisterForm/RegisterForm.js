import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from "react-redux"
import { userRegister } from '../../store/asyncMethods/AuthMethods';
import { Toaster, toast } from "react-hot-toast";
import Store from '../../store';
const RegisterForm = () => {
    const { registerErrors } = Store.getState().AuthReducer;
    const [state, setState] = useState({ name: '', email: '', password: '' })
    const handleChange = e => {
        setState({ ...state, [e.target.name]: e.target.value });
    }
    const handleSubmit = e => {
        e.preventDefault();
        Store.dispatch(userRegister(state));
    }
    useEffect(() => {
        if (registerErrors.length > 0) {
            registerErrors.map(err => toast.error(err.msg))
            Store.dispatch({ type: "CLOSE_REGISTER_ERRORS" })
        }
    }, [registerErrors]);
    return (
        <div className='form' >
            <h1>Register Form</h1>
            <Toaster position="top-center" reverseOrder={false} />
            <form onSubmit={handleSubmit}>
                <input type="text" name='name' onChange={handleChange} placeholder='Enter your Name' />
                <input type="text" name='email' placeholder='Enter your Email' onChange={handleChange} />
                <input type="password" name='password' placeholder='Enter your Password' onChange={handleChange} />
                <input type="submit" placeholder='Register' />
            </form>
        </div>
    )
}

export default RegisterForm