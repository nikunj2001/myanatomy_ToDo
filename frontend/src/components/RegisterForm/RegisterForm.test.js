import React from 'react';
import { shallow, configure } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
import RegisterForm from './RegisterForm';

Enzyme.configure({ adapter: new Adapter() })


describe("LoginForm", () => {
    it("should render my component", () => {
        const wrapper = shallow(<RegisterForm />);
        const tree = wrapper.debug();
        expect(tree).toMatchSnapshot();
    });
});