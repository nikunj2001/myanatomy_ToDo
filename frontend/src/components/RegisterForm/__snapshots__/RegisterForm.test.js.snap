// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTask should render my component 1`] = `
"<div className=\\"form\\">
  <h1>
    Register Form
  </h1>
  <Toaster position=\\"top-center\\" reverseOrder={false} />
  <form onSubmit={[Function: handleSubmit]}>
    <input type=\\"text\\" name=\\"name\\" onChange={[Function: handleChange]} placeholder=\\"Enter your Name\\" />
    <input type=\\"text\\" name=\\"email\\" placeholder=\\"Enter your Email\\" onChange={[Function: handleChange]} />
    <input type=\\"password\\" name=\\"password\\" placeholder=\\"Enter your Password\\" onChange={[Function: handleChange]} />
    <input type=\\"submit\\" placeholder=\\"Register\\" />
  </form>
</div>"
`;

exports[`LoginForm should render my component 1`] = `
"<div className=\\"form\\">
  <h1>
    Register Form
  </h1>
  <Toaster position=\\"top-center\\" reverseOrder={false} />
  <form onSubmit={[Function: handleSubmit]}>
    <input type=\\"text\\" name=\\"name\\" onChange={[Function: handleChange]} placeholder=\\"Enter your Name\\" />
    <input type=\\"text\\" name=\\"email\\" placeholder=\\"Enter your Email\\" onChange={[Function: handleChange]} />
    <input type=\\"password\\" name=\\"password\\" placeholder=\\"Enter your Password\\" onChange={[Function: handleChange]} />
    <input type=\\"submit\\" placeholder=\\"Register\\" />
  </form>
</div>"
`;
